src/azul/chalice.py

```diff
@@ -504,34 +497,122 @@ def swagger_resource(self, file) -> Response:
                                 headers={'Content-Type': content_type},
                                 body=body)
 
-    def threshold(self, *, errors: int, throttles: int):
-        def wrapper(f):
+    # noinspection PyPep8Naming
+    @attr.define(frozen=True)
+    class metric_alarm:
+        """
+        Use this decorator to configure a Lambda's metric alarm, specify a
+        custom number of Lambda invocation resulting in a failure or throttling
+        (threshold) and for which period (5 min - 1 day) are necessary to
+        trigger the metric alarm for the decorated event handler function. This
+        decorator cannot be used to decorate view functions, i.e. functions also
+        decorated with ``@app.route``.
+
+        """
+        #: The AWS/Lambdas namespace for this alarm
+        metric: LambdaMetric
+        #: The value used by the alarm, that when exceeded triggers the alarm
+        threshold: int = attr.ib(default=0)
+        #: The time in seconds to evaluate the alarm threshold
+        period: int = attr.ib(default=5 * 60)
```

Comment:
The grammar in this docstring is very difficult to understand. I've done my best to clear it up but I may have made mistakes, so please review my suggestion carefully for inaccuracies. There is probably room for further improvement here.

```suggestion
        """
        Use this decorator to configure a metric alarm for a Lambda function. This
        decorator cannot be used to decorate view functions, i.e. functions also
        decorated with ``@app.route``.

        """
        #: The AWS/Lambdas namespace for this alarm
        metric: LambdaMetric
        #: The number of failed or throttled lambda invocations that, when exceeded, will trigger the alarm
        threshold: int = attr.ib(default=0)
        #: The interval (in seconds) at which the alarm threshold is evaluated. The default and minimum allowed value is 5 minutes, and the maximum allowed value is 1 day.
        period: int = attr.ib(default=5 * 60)
```